File: 131.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <sstream>
#include <map>
#include <cmath>
#include <algorithm>
#include "pe_utils.hpp"

using namespace std;

typedef long long LL;

vector<LL> cubes;

int main(){
  vector<bool> prime = pe_utils::sieve(1000000);
  LL prev = 1;
  int res = 0;
  for(LL i =2 ; ;++i){
    LL curr = i*i*i;
    if (curr-prev >= 1000000) break; 
    if(prime[curr-prev]) res++;
    prev = curr;
  }
  cout << res << endl;
}

File: 131.py
#define primes
import gmpy
pr = [2]
curr =2 
for i in range(1000):
    pr.append(gmpy.next_prime(curr))
    curr = gmpy.next_prime(curr)

#cubes 
cubes = [i**3 for i in range(1,1000)]
#squares
squares = [i**2 for i in range(1,1000)]

cubes = set(cubes).intersection(squares)

print cubes

for prime in pr:
    for i in cubes:
        if i + prime in cubes:
            print i,prime
            break
